metatensor-core/src/blocks.rs

```diff
@@ -225,7 +225,7 @@ impl TensorBlock {
         if gradient.samples.size() < 1 || gradient.samples.names()[0] != "sample" {
             return Err(Error::InvalidParameter(format!(
                 "'{}' is not valid for the first dimension in the gradients \
-                samples labels, it should be 'sample'", gradient.samples.names()[0]
+                sample labels, it should be 'sample'", gradient.samples.names()[0]
```

Comment:
```suggestion
                samples labels, it should be 'sample'", gradient.samples.names()[0]
```

Since we are using plural everywhere, I'll do it in the docs & error messages as well. I also sounds better for me to consider "the samples dimension" instead of "the sample dimension" since there are multiple samples along this dimension.